contracts/adapters/balancer/BalancerV3RouterAdapter.sol

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Gearbox Protocol. Generalized leverage for DeFi protocols
+// (c) Gearbox Foundation, 2024.
+pragma solidity ^0.8.23;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+import {RAY} from "@gearbox-protocol/core-v3/contracts/libraries/Constants.sol";
+
+import {AbstractAdapter} from "../AbstractAdapter.sol";
+
+import {IBalancerV3Router} from "../../integrations/balancer/IBalancerV3Router.sol";
+import {IBalancerV3Pool} from "../../integrations/balancer/IBalancerV3Pool.sol";
+import {IBalancerV3RouterAdapter} from "../../interfaces/balancer/IBalancerV3RouterAdapter.sol";
+
+/// @title Balancer V3 Router adapter
+/// @notice Implements logic allowing CAs to perform swaps via Balancer V3
+contract BalancerV3RouterAdapter is AbstractAdapter, IBalancerV3RouterAdapter {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    bytes32 public constant override contractType = "AD_BALANCER_V3_ROUTER";
```

Comment:
@Van0k shall be `"ADAPTER::BALANCER_V3_ROUTER"` according to new convention: 

https://github.com/Gearbox-protocol/core-v3/blob/66f9b8e7be833964c935cd43d1e8002b9f08d9ea/contracts/interfaces/base/IAdapter.sol#L11